docs/enhancements/install_logs.md

```diff
@@ -0,0 +1,88 @@
+
+### Goals
+Store OpenShift install logs inside of S3 blob storage.
+
+### Non-Goals
+None
+
+## Open Questions
+1. What is the best way to pass the S3 parameters from the Hive operator to Hive controllers to the OpenShift install pod?
+   1. MUST allow for other blob storage machanisms to be added in the future.
+   1. We'll look to how other projects like Velero do this as a reference.
+1. What object should store the URL of the OpenShift install log in S3?
+
+## Proposal
+Change Hive to store OpenShift install logs in S3 blob storage instead of persistent volumes.
+
+Ensure that future enhancements can use similar mechanisms to upload to other blog storage systems.
+
+## User Stories
+
+### Story 1
+As a Hive administrator,
+I want OpenShift install logs to be stored in S3 compatible blob storage,
+So that simultaneous Hive triggered OpenShift installs can scale with available compute and memory instead of PVs.
+
+### Story 2
+As an Hive Developer,
+I want OpenShift install logs to be stored in S3 compatible blob storage,
+So that while developing, I can use Minio to capture OpenShift install logs.
+
+
+## Implementation Details / Notes / Constraints
+1. Configure S3 URL and credentials in HiveConfig. Have these passed down into OpenShift install pods. From the OpenShift install pods, upload the OpenShift install log to S3.
+  1. If no S3 information is given, do not gather OpenShift install logs.
+1. Remove all PV logic from the Hive code that launches the OpenShift install pod.
+1. Update the log retrieval script to gather the logs from S3 possibly using the URL and credentials in HiveConfig
+1. Update Hive documentation for troubleshooting failed OpenShift installs to point to the OpenShift install logs in S3.
+1. Store the URL of the OpenShift install log in S3 in a TBD Hive object.
+1. Prune the OpenShift install log when the Hive object is removed.
+
+## Risks and Mitigations
+There is a risk of using a large amount of data in S3 and having these OpenShift install logs live forever.
+
+Mitigations:
+- Clean up OpenShift install logs when Hive objects removed
+- End user can remove OpenShift install logs from S3 if desired (won't affect Hive)
+- As a future enhancement, we could explore compression
+
+### Test Plan
+
+- Unit tests for:
+  - ensuring S3 details are added to the controllers pod
+  - ensuring S3 details are added to the OpenShift install pod
+  - ensuring S3 upload is triggered after OpenShift install has completed
+
+- Manual tests for:
+  - ensuring S3 details make it to OpenShift install pod
+  - ensuring OpenShift install log pod uploads OpenShift install log to s3
+  - ensuring OpenShift install log gets removed from s3 when object cleaned up
+
+## Drawbacks
+End user must setup and maintain S3 compatible blob storage.
```

Comment:
An additional drawback, presently we assume a default of gathering logs and putting on pvcs. If you don't have quota/storage classes installs will fail to launch until you explicitly go setup HiveConfig to skipGatherLogs.

In this new mode it sounds more like our default is to not gather and this is a bit of a sketchy thing to roll out on people. I don't want to maintain the pv code as an alternative method though, so I think it's unavoidable. Perhaps worth noting the drawback here though that we're shifting from a default of gather to a default of don't gather.

FWIW ACM disabled log gathering in all their installs due to the pvc issues.

Reply:
Are we planning to deprecate `skipGatherLogs`?